### PR TITLE
Move git_repository into its own Python package and update build.py and test.py to use the new location.

### PR DESCRIPTION
**Pull request by camerski**
Move git_repository from build_workflow into a new "git" package and update build.py and test.py to use the new location.
    
Remove the duplicate GitRepository from test_workflow.
    
Change test.py to use the manifests and git packages. Move bundle_manifest to the manifests package.
    
Rename integ_test.py to integ_test_suite.py.
 
### Testing
Manually verified that `build.py` and `test.py` still work.
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by dblock — APPROVED** [review]
LGTM, consider addressing some minor things below now or later

File: bundle-workflow/python/test_workflow/integ_test_suite.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+class IntegTestSuite:
+    def __init__(self, name, repo):
+        self.name = name
+        self.repo = repo
+
+    def execute(self, cluster):
+        script = self.repo.dir.name + "/integtest.sh"
+        if (os.path.exists(script)):
+            print(f'sh integtest.sh -b {cluster.endpoint()} -p {cluster.port()}')
+#            repo.execute(f'sh integtest.sh -b {cluster.endpoint()} -p {cluster.port()}')
```
[the remark this answers is not in the document]
Comment (camerski) [reply bundle-workflow/python/test_workflow/integ_test_suite.py:12]: Yeah, it's stubbed out for now until we implement the setup-and-tear-down-the-test-cluster code.

**Comment by dblock** [comment]
This needs a rebase.

**Review by mch2 — APPROVED** [review]
lgtm with a rebase

**Pull request merged**